namui/src/namui/manager/web/keyboard_manager.rs

```diff
@@ -23,72 +23,87 @@ impl KeyboardManager {
+                                code_string
+                            );
+                            return;
+                        }
+                        let code = code.unwrap();
+                        pressing_code_set.write().unwrap().insert(code);
+
+                        match code {
+                            Code::Space
+                            | Code::ArrowLeft
+                            | Code::ArrowRight
+                            | Code::ArrowUp
+                            | Code::ArrowDown
+                            | Code::AltLeft
+                            | Code::AltRight => {
+                                event.prevent_default();
+                            }
+                            _ => {}
+                        }
+
+                        crate::event::send(crate::NamuiEvent::KeyDown(crate::KeyEvent { code }));
+                    }
+                }) as Box<dyn FnMut(_)>)
+                .into_js_value()
+                .unchecked_ref(),
+            )
             .unwrap();
 
         document
-            .add_event_listener_with_callback("keyup", key_up_closure.as_ref().unchecked_ref())
+            .add_event_listener_with_callback(
+                "keyup",
+                Closure::wrap(Box::new({
+                    let pressing_code_set = pressing_code_set.clone();
+                    move |event: web_sys::KeyboardEvent| {
+                        let code_string = event.code();
+                        let code = Code::from_str(&code_string);
+                        if code.is_err() {
+                            crate::log!(
+                                "[DEBUG] Fail to get code from key_up callback {}",
+                                code_string
+                            );
+                            return;
+                        }
+                        let code = code.unwrap();
+                        pressing_code_set.write().unwrap().remove(&code);
+                    }
+                }) as Box<dyn FnMut(_)>)
+                .into_js_value()
+                .unchecked_ref(),
+            )
             .unwrap();
 
         ["blur", "visibilitychange"].iter().for_each(|event_name| {
             document
                 .add_event_listener_with_callback(
-                    event_name,
-                    clear_closure.as_ref().unchecked_ref(),
+                    *event_name,
+                    Closure::wrap(Box::new({
+                        let pressing_code_set = pressing_code_set.clone();
+                        move || {
+                            pressing_code_set.write().unwrap().clear();
+                        }
+                    }) as Box<dyn FnMut()>)
+                    .into_js_value()
+                    .unchecked_ref(),
                 )
                 .unwrap();
         });
 
-        key_down_closure.forget();
-        key_up_closure.forget();
-        clear_closure.forget();
```

Comment:
No need to forget?

Reply:
Aaaah, [into_js_value()](https://github.com/NamseEnt/namseent/pull/206/files#diff-50fc3d0dfc2f8c38f6aedf28ccdbc3f1b3c07ff01d054e56469184fc4f3cd5b3R101)